### ISSUE TITLE
New hyperdrive inspiration

### ISSUE THREAD

**Issue by ProNooberEX** [endless-sky/endless-sky#7067]
### Problem Description

It has come to my attention that there are only three types of hyperdrives in the game; the hyperdrive, scramdrive and jump drive. In comparison, many other outfitters had close to ten types of choice to average gamers. With the developing and perfecting world of endless sky, such limitation on the type of jump drives will serverly hinder the overall development of the world view of endless sky and later alien species, whom might need to use new drive to travel around or mention it in their history when the gamer arrived.  

### Related Issue Links

*

### Desired Solution

The introduction of a new drive will open up new possiblities for new groups of government, missions, storylines and of course new combination of outfitters. Under this circumstance, I decided to post a discussion for suggestions on new types of drives that could be put into the game based on the current background of technology and scientific basis of endless sky where everyone can put his or her ideas in words or picture to provide Mr Micheal for new possible choices of traveling between the stars under the establishment of the hyperlink background.
Personally, a good example of a new drive will be based on a science or science fiction novel based on existent theoretical physics. The one drive I wish to introduce in person is the "luminescence" drive, orginating from the idea of super and sub luminescent space in the sci-fi book "to sleep in a see of stars", in the end of the book, the author mentions the mechanics behind a hyperdrive that carries the humans between solar systems by creating a space-time bubble in which the ship enters a "super illuminant" space where ships can essentially break light speeds without disobeying the normal laws of Einsteinien physics in sub-illuminant, or normal space time. To explain this simply, take minecraft's nether realm as an example. Nether composes of a different game mechanic compared to the overworld, behaviors or traits that violate game laws in the overworld might be viewed as normal and acceptbale behavior or traits by the game in the nether. Lava is a good difference to illustrate; in the overworld, lava flows very slowly, a few hundred ticks per block of movement by lava, while in the nether, lava flows only tens of ticks per block. Here, the speed in which lava flows in the nether will never be achieved in the overworld(without cheats of course), but suppose a player puts lava in the nether, will it affect the overworld? Apparently not. The drive follows the same mechanic, a drive simply puts the ship in a realm where light speed isn't the limiting speed, just as in the nether world, one block of movement by a player is the equivalent of seven blocks of movement in the overworld. Following the same logic, the luminescence drive can allow ships access to this "nether" in space-time and, in turn, reaching far greater speed every capable in normal space and enable fast traveling between the stars.

### Alternative Approaches

*

### Additional Context

*

**Comment by ProNooberEX** [comment]
Statistics suggestions for the drive:
1. As this drive is essentially just making the ship travel at normal speeds but in a different space-time, its fuel consumption will probably lower than the hyperdrive.
2. This drive needs the ship to move through this new space-time using its own battery power, therefore it probably will deduct a certain amount of battery power after the jump.

**Comment by ProNooberEX** [comment]
P.S: for the first one, the "fuel" is hyperspace fuel while the second one is battery energy

**Comment by Zitchas** [comment]
Having additional kinds of interstellar travel would, indeed, be interesting to have. However, there are a couple of caveats for doing so within the Endless Sky context.

1. The mechanics of the game are structured around the current hyperdrive system. Most alternate systems (such as warp drives from Star Trek, Slip Stream drives from Andromeda, hyperspace travel from Babylon 5, or, from the sounds of it, this "luminescence" drive.) tend to have the users navigating either in interstellar space, or navigating in an alternate dimension. While either of these options would be very interesting to have in Endless Sky, they would necessitate an extremely large amount of work to add. (both the new realm to navigate in, exit/entry mechanics, and working out how it lines up with the current in-game mechanics such as time.)
2. Mechanically speaking, we currently have something of a block in place regarding alternate forms of interstellar drives. This is due to the story; specifically scenes that refer to the player's choice of drives. Every additional drive will need to add a new branch to those scenes for dialog covering that variety. As such, there's been a moratorium on new drives (regardless of the lore) until we have the mechanics to support outfit variants, which would let us group drives together for testing purposes.

**Comment by ProNooberEX** [comment]
Indeed, the above challenges could be very tricky, therefore I will try to modify my idea to accustom it to the current hyperlink system and jump mechanics, thanks for the advice!

**Comment by ProNooberEX** [comment]
Modifications for the new drive: 
1. The new drive will still follow the same rules for the normal hyperdrive(jumping from system to system one at a time within existing hyperlinks), but each jump location will be located on the star of each system(as the gravity pit is greatest there in each system) and only half a block of fuel will be consumed on every jump with a quarter of energy of battery consumed as well.
2. The new drive, personally, probably will be placed in the Future Quarg storyline(as personally, the Quarg "who always pursues efficeny" and "looks down on Korath propulsion with distaste", will be intended to develop such a engine that can allow them to monitor other species, both human and aliens, with less fuel consumptions than their jump drives, especially within systems of species that are advanced enough to develop hyperlinks on their own)

**Comment by ira210** [comment]
The problem is that, for example in the free worlds missions, there are missions which check for what drive you have installed on your ship and change dialogue based on that and because of the way mission syntax works is new drives were added those mission wouldn’t fire so the fw campaign couldn’t be completed unless new missions were added for these new drives.

**Comment by tibetiroka** [comment]
Currently, there are two main routes we can take when adding new drives:
 - create a slightly different hyperdrive, that has different fuel consumption or velocity tolerance, or
 - create a new "link type" between systems that the drive would utilize.

The first one doesn't really merit an open issue, and it's also better to keep things relatively consistent. The second one would require major code work and map changes to utilize, which doesn't feel needed in the game.

In addition, maintaining game balance with the new drive types would be more difficult, and, like others said, many existing missions would need to be changed to incorporate the new possibilities. That's a frequent struggle even with jump drives.
